### ISSUE TITLE
Data Reconciliation not catching error / aborting in testsuite

### ISSUE THREAD

**Issue by AnHeuermann** [OpenModelica/OpenModelica#10109]
### Description

Data reconcilliation is evaluating the DAE which can fail from time to time, e.g. when a non-linear equation system can't be solved or if some equation is dividing by zero. Currently these errors are not handled and the simulation will abort.

In https://github.com/OpenModelica/OpenModelica/pull/10063 the example [testsuite/openmodelica/dataReconciliation/TSP_Pipe3](https://github.com/OpenModelica/OpenModelica/blob/master/testsuite/openmodelica/dataReconciliation/TSP_Pipe3.mos#L2264-L2266) adds such an error with a segmentation fault (sometimes, not on my machine....) to our testsuite, but there was already a similar error in the testsuite in [testsuite/openmodelica/dataReconciliation/TSP_FourFlows4.mos](https://github.com/OpenModelica/OpenModelica/blob/master/testsuite/openmodelica/dataReconciliation/TSP_FourFlows4.mos#L3694-L3696).

### Steps to Reproduce

Run [https://github.com/OpenModelica/OpenModelica/blob/master/testsuite/openmodelica/dataReconciliation/TSP_Pipe3.mos](testsuite/openmodelica/dataReconciliation/TSP_Pipe3.mos).

### Expected Behavior

Use `MMC_TRY_INTERNAL` and handle errors when a function inside the DAE failes.

```C
/* try */
#if !defined(OMC_EMCC)
  MMC_TRY_INTERNAL(simulationJumpBuffer)
#endif

/* Evaluate DAE */
[...]
data->callback->functionDAE(data, threadData);
[...]

/* catch*/
#if !defined(OMC_EMCC)
  MMC_CATCH_INTERNAL(simulationJumpBuffer)
#endif

/* Handle errors */
```

### Version and OS

 - OpenModelica Version: v1.21.0-dev-216-ge469b0cb37-cmake
 - OS: Ubuntu 22.04.1 LTS
 - Versions of used Modelica libraries if applicable: ThermoSysPro v3.2.0
 
 -----------------

Adding @phannebohm as well.

**Comment by arun3688** [comment]
@AnHeuermann Can we use the `MMC_TRY_INTERNAL` to catch the errors in Jacobins for example'

```
MMC_TRY_INTERNAL(simulationJumpBuffer)

ANALYTIC_JACOBIAN *jacobian = &(data->simulationInfo->analyticJacobians[index]);
data->callback->initialAnalyticJacobianF(data, threadData, jacobian);

MMC_CATCH_INTERNAL(simulationJumpBuffer)
```

**Comment by AnHeuermann** [comment]
I guess, but I never tried. I also didn't test my proposed solution at the top. But something in this direction should work. It is only a question on what to do when encountering an error.

**Comment by arun3688** [comment]
@AnHeuermann that is true there are some really hard to catch errors, but i will try your proposed solution and see

**Comment by audrey-jardin** [comment]
@adrpo , @arun3688 , @bmazurie : we could take the opportunity of the ongoing DFD contract about "data reconciliation" to solve the ticket?
